### PR TITLE
Chore use offical redux typing

### PR DESCRIPTION
**Pull request by e-schultz**
@wbuchwalter @bennett000 redux 3.4 has official typings along with it now. Think the wrapActionCreators probably needs a  little more work though

### CODE REVIEW

**Comment by e-schultz** [comment]
@wbuchwalter @bennett000 - moving towards using the official redux types + tests for the observable store, and so I could use some of the fakeAsync to make RxJs testing easier, move from mocha to jasmine to use the angular2 testing utilities. Still a work in progress.

**Comment by wbuchwalter** [comment]
Awesome work guys!
Glad to see so much progress 🎉

File: src/components/provider.ts [hunk 1]
```diff
@@ -3,13 +3,15 @@ import Connector from './connector';
...
   const _connector = new Connector(store);
   const factory = (): INgRedux<T> => <INgRedux<T>>Object.assign({},
-    { connect: _connector.connect,
+    {
+      connect: _connector.connect,
       mapDispatchToTarget: _connector.mapDispatchToTarget
     },
     store);
-  return provide('ngRedux', { useFactory: factory });
+  return provide(NgRedux, { useFactory: () => new NgRedux(factory()) });
```
Comment (e-schultz) [inline src/components/provider.ts]: @wbuchwalter @bennett000

Wasn't a fan of needing to have the mix of `@Inject`/etc - this lets us just deal with registering one provider.

**Comment by e-schultz** [comment]
@wbuchwalter @bennett000 - thinking this is getting close to a point of being mergeable - will squash/clean up the commits as I have quite a bit of noisy 'wip' things in here.

Simplified the provide a bit, and also made it so we didn't need to have both connect.ts and ng-redux.ts - realized ng-redux.ts had lots that was just re-exposing the connect, and didn't seem to make sense to have two separate classes.

**Comment by e-schultz** [comment]
closing this pr, submitting another one with a cleaned up commit history - didn't want to squash this branch just so I could have some reference points for myself in the future.

**Pull request closed without merging**